File: contests/arc004/arc004_a.py
# A - 2点間距離の最大値 ( The longest distance )
# https://atcoder.jp/contests/arc004/submissions/14200487

N = int(input())

arr_x = []
arr_y = []
arr_dist = []

for i in range(0, N):
    x, y = list(map(int, input().split()))
    arr_x.append(x)
    arr_y.append(y)

for i in range(0, N):
    for j in range(i, N):
        arr_dist.append((arr_x[j]-arr_x[i])**2 + (arr_y[j]-arr_y[i])**2)

print(max(arr_dist) ** 0.5)

File: contests/arc004/arc004_a.cpp
// A - 2点間距離の最大値 ( The longest distance )
// https://atcoder.jp/contests/arc004/tasks/arc004_1

#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int n; cin >> n;
    int a[105], b[105];
    for (int i=0;i<n;i++) {
        cin >> a[i] >> b[i];
    }

    double t = 0;
    for(int i=0;i<n;i++) {
        for(int j=0;j<n;j++) {
            double dist = sqrt((a[i]-a[j])*(a[i]-a[j]) + (b[i]-b[j])*(b[i]-b[j]));
            t = max(t, dist);
        }
    }

    cout << t << endl;
}
